### ISSUE TITLE
Dashboard doesn't show errors in routers in case of failure!!

### ISSUE THREAD

**Issue by MohamedMSaeed** [containous/traefik#6275]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?
I use Traefik dashboard to list failed Routers. Router should give error when service is not found for example but it doesn't. 

### What did you expect to see?
Iist errors in routers.

### What did you see instead?
Failed routers are not listed as error.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
2.1.3
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
I use traefik with kubernetes

#### Logs:
```
[traefik-756ddf7997-mg6rp] time="2020-02-04T13:34:30Z" level=error msg="kubernetes service not found: do/foo" providerName=kubernetescrd ingress=foo-www-https namespace=do
[traefik-756ddf7997-mg6rp] time="2020-02-04T13:34:30Z" level=error msg="kubernetes service not found: do/foo" providerName=kubernetescrd ingress=foo-www-https namespace=do
[traefik-756ddf7997-mg6rp] time="2020-02-04T13:34:32Z" level=error msg="kubernetes service not found: do/foo" providerName=kubernetescrd ingress=foo-www-https namespace=do
[traefik-756ddf7997-mg6rp] time="2020-02-04T13:34:32Z" level=error msg="kubernetes service not found: do/foo" ingress=foo-www-https providerName=kubernetescrd namespace=do
[traefik-756ddf7997-mg6rp] time="2020-02-04T13:34:32Z" level=error msg="kubernetes service not found: do/foo" namespace=do providerName=kubernetescrd ingress=foo-www-https
```

#### UI snapshot:
![image](https://user-images.githubusercontent.com/35697318/73749527-3cc3fb80-4764-11ea-875c-c5d655994107.png)

#### Scenario:

Let's say that we use the following ingressrouter to route any request "foo.com" to service "foo"

IngressRoute
```yml
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRoute
metadata:
  annotations:
    kubernetes.io/ingress.class: traefik
  name: foo-www-https
  namespace: do
spec:
  entryPoints:
    - websecure
  routes:
  - kind: Rule
    match: Host(`foo.com`) && PathPrefix(`/`)
    priority: 0
    services:
    - name: foo
      port: 80
```

In traefik version 2.0.7, if the service "foo" was not found, the router will be listed as error. If it was there the router will be listed as success. 

In traefik version 2.1.3 if the service is defined, router will be listed in success. If I stopped the service "foo" the router will be removed from success and will not be listed in errors.

#### values.json

I deployed traefik using helm. here is values.json

<details>
<summary>values.json</summary>

```yml
# Default values for Traefik
image:
  name: traefik
  tag: 2.1.3

#
# Configure the deployment
#
deployment:
  # Number of pods of the deployment
  replicas: 1

rollingUpdate:
  maxUnavailable: 1
  maxSurge: 1

#
# Configure Traefik entry points
# Additional arguments to be passed at Traefik's binary
## Use curly braces to pass values: `helm install --set="{--providers.kubernetesingress,--global.checknewversion=true}" ."
additionalArguments:
  - "--entryPoints.web.forwardedHeaders.insecure"
  - "--entryPoints.websecure.forwardedHeaders.insecure"
  - "--api.insecure=true"
  - "--accesslog=true"
  - "--tracing=true"
  - "--tracing.serviceName=traefik"
  - "--tracing.spanNameLimit=1500000"
  - "--tracing.datadog=true"
  - "--tracing.datadog.localAgentHostPort=100.96.2.230:8126"
  - "--metrics=true"
  - "--metrics.datadog=true"
  - "--metrics.datadog.address=100.96.2.230:8125"
  - "--metrics.datadog.addServicesLabels=true"
  - "--metrics.datadog.pushInterval=10s"

# Environment variables to be passed to Traefik's binary
env: {}
#  - name: SOME_VAR
#    value: some-var-value
#  - name: SOME_OTHER_VAR
#    value: some-other-var-value

#
ports:
  # The name of this one can't be changed as it is used for the readiness and
  # liveness probes, but you can adjust its config to your liking
  traefik:
    port: 9000
    # Defines whether the port is exposed if service.type is LoadBalancer or
    # NodePort.
    #
    # You SHOULD NOT expose the traefik port on production deployments.
    # If you want to access it from outside of your cluster,
    # use `kubectl proxy` or create a secure ingress
    expose: false
    # The exposed port for this service
    exposedPort: 9000
  web:
    port: 8000
    expose: true
    exposedPort: 80
    # Use nodeport if set. This is useful if you have configured Traefik in a
    # LoadBalancer
    # nodePort: 32080
  websecure:
    port: 8443
    expose: true
    exposedPort: 443
  # nodePort: 32443

# Options for the main traefik service, where the entrypoints traffic comes
# from.
service:
  type: LoadBalancer
  # Additional annotations (e.g. for cloud provider specific config)
  annotations: {}
  # Additional entries here will be added to the service spec. Cannot contains
  # type, selector or ports entries.
  spec: {}
    # externalTrafficPolicy: Cluster
    # loadBalancerIP: "1.2.3.4"
    # clusterIP: "2.3.4.5"

dashboard:
  # Enable the dashboard on Traefik
  enable: true

  # Expose the dashboard and api through an ingress route at /dashboard
  # and /api This is not secure and SHOULD NOT be enabled on production
  # deployments
  ingressRoute: true

logs:
  loglevel: DEBUG
#
resources:
   requests:
     cpu: "100m"
     memory: "50Mi"
   limits:
     cpu: "300m"
     memory: "150Mi"
affinity: {}
nodeSelector: {}
tolerations: []

```

</details>

**Comment by MohamedMSaeed** [comment]
It was closed by mistake

**Comment by ldez** [comment]
Hello,

In the issue template, we ask for configuration, logs, etc., could you provide those information.

If a router cannot be created because of an invalid definition, it's not possible to get information in the UI because something that cannot be created cannot be associated to an error.

**Comment by MohamedMSaeed** [comment]
Hi,

## Logs:
```
[traefik-756ddf7997-mg6rp] time="2020-02-04T13:34:30Z" level=error msg="kubernetes service not found: do/foo" providerName=kubernetescrd ingress=foo-www-https namespace=do
[traefik-756ddf7997-mg6rp] time="2020-02-04T13:34:30Z" level=error msg="kubernetes service not found: do/foo" providerName=kubernetescrd ingress=foo-www-https namespace=do
[traefik-756ddf7997-mg6rp] time="2020-02-04T13:34:32Z" level=error msg="kubernetes service not found: do/foo" providerName=kubernetescrd ingress=foo-www-https namespace=do
[traefik-756ddf7997-mg6rp] time="2020-02-04T13:34:32Z" level=error msg="kubernetes service not found: do/foo" ingress=foo-www-https providerName=kubernetescrd namespace=do
[traefik-756ddf7997-mg6rp] time="2020-02-04T13:34:32Z" level=error msg="kubernetes service not found: do/foo" namespace=do providerName=kubernetescrd ingress=foo-www-https
```

## UI snapshot:
![image](https://user-images.githubusercontent.com/35697318/73749527-3cc3fb80-4764-11ea-875c-c5d655994107.png)


## Scenario:

Let's say that we use the following ingressrouter to route any request "foo.com" to service "foo"

IngressRouter
```
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRoute
metadata:
  annotations:
    kubernetes.io/ingress.class: traefik
  name: foo
spec:
  entryPoints:
    - websecure
  routes:
  - kind: Rule
    match: Host(`foo.com`) && PathPrefix(`/`)
    priority: 0
    services:
    - name: foo
      port: 80
```
In traefik version 2.0.7, if the service "foo" was not found, the router will be listed as error. If it was there the router will be listed as success. 

In traefik version 2.1.3 if the service is defined, router will be listed in success. If I stopped the service "foo" the router will be removed from success and will not be listed in errors.

**Comment by MohamedMSaeed** [comment]
I deployed traefik using helm, should I post the var file?

**Comment by MohamedMSaeed** [comment]
values.json
```
# Default values for Traefik
image:
  name: traefik
  tag: 2.1.3

#
# Configure the deployment
#
deployment:
  # Number of pods of the deployment
  replicas: 1

rollingUpdate:
  maxUnavailable: 1
  maxSurge: 1

#
# Configure Traefik entry points
# Additional arguments to be passed at Traefik's binary
## Use curly braces to pass values: `helm install --set="{--providers.kubernetesingress,--global.checknewversion=true}" ."
additionalArguments:
  - "--entryPoints.web.forwardedHeaders.insecure"
  - "--entryPoints.websecure.forwardedHeaders.insecure"
  - "--api.insecure=true"
  - "--accesslog=true"
  - "--tracing=true"
  - "--tracing.serviceName=traefik"
  - "--tracing.spanNameLimit=1500000"
  - "--tracing.datadog=true"
  - "--tracing.datadog.localAgentHostPort=100.96.2.230:8126"
  - "--metrics=true"
  - "--metrics.datadog=true"
  - "--metrics.datadog.address=100.96.2.230:8125"
  - "--metrics.datadog.addServicesLabels=true"
  - "--metrics.datadog.pushInterval=10s"

# Environment variables to be passed to Traefik's binary
env: {}
#  - name: SOME_VAR
#    value: some-var-value
#  - name: SOME_OTHER_VAR
#    value: some-other-var-value

#
ports:
  # The name of this one can't be changed as it is used for the readiness and
  # liveness probes, but you can adjust its config to your liking
  traefik:
    port: 9000
    # Defines whether the port is exposed if service.type is LoadBalancer or
    # NodePort.
    #
    # You SHOULD NOT expose the traefik port on production deployments.
    # If you want to access it from outside of your cluster,
    # use `kubectl proxy` or create a secure ingress
    expose: false
    # The exposed port for this service
    exposedPort: 9000
  web:
    port: 8000
    expose: true
    exposedPort: 80
    # Use nodeport if set. This is useful if you have configured Traefik in a
    # LoadBalancer
    # nodePort: 32080
  websecure:
    port: 8443
    expose: true
    exposedPort: 443
  # nodePort: 32443

# Options for the main traefik service, where the entrypoints traffic comes
# from.
service:
  type: LoadBalancer
  # Additional annotations (e.g. for cloud provider specific config)
  annotations: {}
  # Additional entries here will be added to the service spec. Cannot contains
  # type, selector or ports entries.
  spec: {}
    # externalTrafficPolicy: Cluster
    # loadBalancerIP: "1.2.3.4"
    # clusterIP: "2.3.4.5"

dashboard:
  # Enable the dashboard on Traefik
  enable: true

  # Expose the dashboard and api through an ingress route at /dashboard
  # and /api This is not secure and SHOULD NOT be enabled on production
  # deployments
  ingressRoute: true

logs:
  loglevel: DEBUG
#
resources:
   requests:
     cpu: "100m"
     memory: "50Mi"
   limits:
     cpu: "300m"
     memory: "150Mi"
affinity: {}
nodeSelector: {}
tolerations: []

```

**Comment by ldez** [comment]
Could you editing the issue description to add those elements.

**Comment by MohamedMSaeed** [comment]
I edited the issue description. If there is any missing information, please tell me. 

Thanks

**Comment by jbdoumenjou** [comment]
Hi @MohamedMSaeed,

The error displayed in dashboard with traefik v2.0 was due to a bug resulting in a (non-functional) router being generated, that has been fixed in the v2.1. The errors information is available in the log.
However, there is indeed a bug in the error log when a kubernetes service is not found, it will be fixed in a PR coming soon.

**Comment by traefiker** [comment]
Closed by #6311.
